src/libmongoc/tests/test-mongoc-sdam.c

```diff
@@ -215,6 +216,250 @@ test_sdam_cb (bson_t *test)
+ */
+static void
+deactivate_failpoints_on_all_servers (mongoc_client_t *client)
+{
+   int i;
+   uint32_t server_id;
+   mongoc_set_t *servers;
+   bson_t cmd;
+   bson_error_t error;
+
+   bson_init (&cmd);
+   BCON_APPEND (&cmd, "configureFailPoint", "failCommand", "mode", "off");
+
+   servers = client->topology->description.servers;
+
+   for (i = 0; i < servers->items_len; i++) {
+      bool ret;
+
+      server_id = servers->items[i].id;
+      ret = mongoc_client_command_simple_with_server_id (client,
+                                                         "admin",
+                                                         &cmd,
+                                                         NULL /* read prefs */,
+                                                         server_id,
+                                                         NULL /* reply */,
+                                                         &error);
+      if (!ret) {
+         MONGOC_DEBUG ("error disabling failpoint: %s", error.message);
+      }
+   }
+
+   bson_destroy (&cmd);
+}
+
+static void
+run_one_integration_test (json_test_config_t *config, bson_t *test)
+{
+   json_test_ctx_t ctx;
+   json_test_ctx_t thread_ctx[2];
+   bson_error_t error;
+   mongoc_client_pool_t *pool;
+   mongoc_client_t *setup_client;
+   mongoc_client_t *test_client;
+   mongoc_client_t *thread_clients[2];
+   const char *db_name;
+   const char *coll_name;
+   mongoc_uri_t *uri;
+
+   MONGOC_DEBUG ("running test: %s", bson_lookup_utf8 (test, "description"));
+
+   uri = test_framework_get_uri ();
+   if (bson_has_field (test, "clientOptions")) {
+      bson_t client_opts;
+
+      bson_lookup_doc (test, "clientOptions", &client_opts);
+      set_uri_opts_from_bson (uri, &client_opts);
+   }
+   mongoc_uri_set_option_as_bool (uri, MONGOC_URI_DIRECTCONNECTION, false);
+
+   db_name = bson_lookup_utf8 (config->scenario, "database_name");
+   coll_name = bson_lookup_utf8 (config->scenario, "collection_name");
+
+   /* SDAM integration tests require streamable ismaster support, which is only
+    * available for a client pool. */
+   pool = mongoc_client_pool_new (uri);
+   mongoc_client_pool_set_error_api (pool, MONGOC_ERROR_API_VERSION_2);
+   test_framework_set_pool_ssl_opts (pool);
+
+   setup_client = test_framework_client_new ();
+   /* Disable failpoints that may have been enabled in a previous test run. */
+   deactivate_failpoints_on_all_servers (setup_client);
+   mongoc_client_command_simple (setup_client,
+                                 "admin",
+                                 tmp_bson ("{'killAllSessions': []}"),
+                                 NULL,
+                                 NULL,
+                                 &error);
+
+   insert_data (db_name, coll_name, config->scenario);
+
+   if (bson_has_field (test, "failPoint")) {
+      activate_fail_point (setup_client, 0, test, "failPoint");
+   }
```

Comment:
AFAIK, this is for the top-level field. Do any of the SDAM tests actually use this, or is it just covering the bases since the SDAM spec readme alludes to the Transactions test format?